command/debug/debug.go

```diff
@@ -320,212 +323,218 @@ func (c *cmd) captureStatic() error {
 	for output, v := range outputs {
 		marshaled, err := json.MarshalIndent(v, "", "\t")
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 
 		err = ioutil.WriteFile(fmt.Sprintf("%s/%s.json", c.output, output), marshaled, 0644)
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 	}
 
-	return errors
+	return errs
 }
 
-// captureDynamic blocks for the duration of the command
+// captureInterval blocks for the duration of the command
 // specified by the duration flag, capturing the dynamic
 // targets at the interval specified
-func (c *cmd) captureDynamic() error {
-	successChan := make(chan int64)
-	errCh := make(chan error)
+func (c *cmd) captureInterval() error {
+	intervalChn := time.NewTicker(c.interval)
+	defer intervalChn.Stop()
 	durationChn := time.After(c.duration)
 	intervalCount := 0
 
 	c.UI.Output(fmt.Sprintf("Beginning capture interval %s (%d)", time.Now().Local().String(), intervalCount))
 
-	// We'll wait for all of the targets configured to be
-	// captured before continuing
-	var wg sync.WaitGroup
-
-	capture := func() {
-		timestamp := time.Now().Local().Unix()
-
-		// Make the directory that will store all captured data
-		// for this interval
-		timestampDir := fmt.Sprintf("%s/%d", c.output, timestamp)
-		err := os.MkdirAll(timestampDir, 0755)
-		if err != nil {
-			errCh <- err
+	err := capture(c)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case t := <-intervalChn.C:
+			intervalCount++
+			c.UI.Output(fmt.Sprintf("Capture successful %s (%d)", t.Local().String(), intervalCount))
+			err := capture(c)
+			if err != nil {
+				return err
+			}
+		case <-durationChn:
+			intervalChn.Stop()
+			return nil
+		case <-c.shutdownCh:
+			return errors.New("stopping collection due to shutdown signal")
 		}
+	}
+}
 
-		// Capture metrics
-		if c.configuredTarget("metrics") {
-			wg.Add(1)
+func capture(c *cmd) error {
```

Comment:
I think this could be renamed to something like captureShortLived or captureSnapshots, to contrast with captureLongRunning. What do you think?

Reply:
I ran out of func names starting with capture for that point.
I like `captureShortLived`, I will make the change.